programs/psa/psa_hash.c

```diff
@@ -0,0 +1,144 @@
+/*
```

Comment:
It's unfortunate that we don't run sample programs on the CI. I would like that at least for _new_ programs we arrange to run them on the CI. https://github.com/Mbed-TLS/mbedtls/pull/2698 sets up some infrastructure to make that easier. Can we rework this as a part of the epic to get the PSA demo programs in?

Reply:
As discussed today, I'm going to split https://github.com/Mbed-TLS/mbedtls/pull/2698 into two parts: one to add `run_demos.py` to CI runs, and one for `cert_write_demo.hs`. We'll push the `run_demos.py` part quickly and we'll add a demo script for the new sample programs.

Reply:
https://github.com/Mbed-TLS/mbedtls/pull/7982 is available for review. I would like to merge it and the hash demo independently, but to make a follow-up to run the hash demo on the CI before we merge the other demos.